homeassistant/components/camera/amcrest.py

```diff
@@ -64,26 +82,84 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
             notification_id=NOTIFICATION_ID)
         return False
 
-    add_devices([AmcrestCam(config, data)])
+    add_devices([AmcrestCam(hass, config, data)])
     return True
 
 
 class AmcrestCam(Camera):
     """An implementation of an Amcrest IP camera."""
 
-    def __init__(self, device_info, data):
+    def __init__(self, hass, device_info, data):
         """Initialize an Amcrest camera."""
         super(AmcrestCam, self).__init__()
+        self._base_url = '%s://%s:%s/cgi-bin' % (
```

Comment:
Convention in Python is that variables starting with an `_` are considered private.